core/example/benchmark/Cabana_CommPerformance.cpp

```diff
@@ -0,0 +1,586 @@
+            Cabana::Halo<comm_memory_space> halo_general(
+                comm, num_particle, comm_export_ids, comm_export_ranks );
+            halo_general_create.stop(fraction);
+
+            // Resize the particles for gather.
+            particles.resize( halo_fast.numLocal() + halo_fast.numGhost() );
+
+            // Gather the aosoa as a whole. Do host/device copies as needed.
+            halo_aosoa_gather.start(fraction);
+            auto comm_particles =
+                Cabana::Experimental::create_mirror_view_and_copy(
+                    comm_memory_space(), particles );
+            Cabana::gather( halo_fast, comm_particles );
+            Cabana::deep_copy( particles, comm_particles );
+            halo_aosoa_gather.stop(fraction);
+
+            // Gather the aosoa using individual slices.
+            halo_slice_gather.start(fraction);
+
+            comm_particles =
+                Cabana::Experimental::create_mirror_view_and_copy(
+                    comm_memory_space(), particles );
+
+            auto s0 = comm_particles.template slice<0>();
+            Cabana::gather( halo_fast, s0 );
+
+            auto s1 = comm_particles.template slice<1>();
+            Cabana::gather( halo_fast, s1 );
+
+            auto s2 = comm_particles.template slice<2>();
+            Cabana::gather( halo_fast, s2 );
+
+            auto s3 = comm_particles.template slice<3>();
+            Cabana::gather( halo_fast, s3 );
+
+            Cabana::deep_copy( particles, comm_particles );
+
+            halo_slice_gather.stop(fraction);
+
+            // Scatter the aosoa using individual slices.
+            halo_slice_scatter.start(fraction);
+
+            comm_particles =
+                Cabana::Experimental::create_mirror_view_and_copy(
+                    comm_memory_space(), particles );
+
+            s0 = comm_particles.template slice<0>();
+            Cabana::scatter( halo_fast, s0 );
+
+            s1 = comm_particles.template slice<1>();
+            Cabana::scatter( halo_fast, s1 );
+
+            s2 = comm_particles.template slice<2>();
+            Cabana::scatter( halo_fast, s2 );
+
+            s3 = comm_particles.template slice<3>();
+            Cabana::scatter( halo_fast, s3 );
+
+            Cabana::deep_copy( particles, comm_particles );
+            halo_slice_scatter.stop(fraction);
+        }
+    }
+
+    // Output results.
+    halo_fast_create.outputResults( file, "send_bytes", comm_bytes );
+    halo_general_create.outputResults( file, "send_bytes", comm_bytes );
+    halo_aosoa_gather.outputResults( file, "send_bytes", comm_bytes );
+    halo_slice_gather.outputResults( file, "send_bytes", comm_bytes );
+    halo_slice_scatter.outputResults( file, "send_bytes", comm_bytes );
+}
+
+//---------------------------------------------------------------------------//
+// main
+int main( int argc, char* argv[] )
+{
+    // Initialize environment
+    MPI_Init( &argc, &argv );
+    Kokkos::initialize( argc, argv );
+
```

Comment:
Exit here if `argc < 3` and consider printing usage

Reply:
done